src/overlays/actors/ovl_Bg_Ladder/z_bg_ladder.c

```diff
@@ -19,22 +37,129 @@ const ActorInit Bg_Ladder_InitVars = {
     (ActorFunc)BgLadder_Init,
     (ActorFunc)BgLadder_Destroy,
     (ActorFunc)BgLadder_Update,
-    (ActorFunc)BgLadder_Draw
+    (ActorFunc)BgLadder_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Ladder_0x80A4D9F0/BgLadder_Init.asm")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static u32 sLadderDLists[] = {
+    object_ladder_12rung_dlist,
+    object_ladder_16rung_dlist,
+    object_ladder_20rung_dlist,
+    object_ladder_24rung_dlist,
+};
+
+void BgLadder_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgLadder* this = THIS;
+    BgLadderSize size;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+
+    // params is treated as two bytes: switchFlag, size
+    this->switchFlag = (thisx->params >> 8) & 0xFF;
+    thisx->params &= 0xFF;
```

Comment:
Does `this->dyna.actor.params` not match? If `thisx` is required to match, a comment stating such would be nice.

Reply:
It's weird, I could make 1 of the 3 lines into `this->dyna.actor.params`, but not all 3 (or 2). 
It seemed _worse_ to mix the two styles so I left them all `thisx->params`? I can add a comment though